scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -199,6 +199,24 @@ sealed trait Execution[+T] extends java.io.Serializable { self: Product =>
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
 
+  /**
+   * Add an extra config for current execution.
+   * This changes of config are isolated and affect only current execution.
+   * All changes will disappear for any next operation on execution.
+   */
+  def addConfig(extraConfig: Config): Execution[T] =
```

Comment:
Do you think we need this method? Since this only "sticks" for just 1 execution, maybe using `Execution.withConfig` is more clear? I'm worried this makes it looks like you could do:

```
myExecuiton
  .addConfig(a)
  .addConfig(b)
  .addConfig(c)
```
 and get a, b, and c all added in the result, when in fact a and b get ignored.